restcomm/restcomm.dao/src/main/java/org/restcomm/connect/dao/mybatis/MybatisExtensionsConfigurationDao.java

```diff
@@ -276,4 +283,86 @@ private ExtensionConfiguration toExtensionConfiguration(final Map<String, Object
         map.put("enabled", extensionConfiguration.isEnabled());
         return map;
     }
+
+    @Override
+    public ExtensionConfiguration getAccountExtensionConfiguration(String accountSid, String extensionSid) {
+        final SqlSession session = sessions.openSession();
+        ExtensionConfiguration extensionConfiguration = null;
+        try {
+            Map<String, Object> params = new HashMap<String, Object>();
+            params.put("account_sid", accountSid.toString());
+            params.put("extension_sid", extensionSid.toString());
+            final Map<String, Object> result  = session.selectOne(namespace + "getAccountExtensionConfiguration", params);
+            if (result != null) {
+                extensionConfiguration = toAccountsExtensionConfiguration(result);
+            }
+            return extensionConfiguration;
+        } finally {
+            session.close();
+        }
+    }
+
+    @Override
+    public void addAccountExtensionConfiguration(ExtensionConfiguration extensionConfiguration, Sid accountSid) throws ConfigurationException {
+        final SqlSession session = sessions.openSession();
+        try {
+            if (extensionConfiguration != null && extensionConfiguration.getConfigurationData() != null) {
+                if (validate(extensionConfiguration)) {
+                    final Map<String, Object> map = new HashMap<String, Object>();
+                    map.put("account_sid", DaoUtils.writeSid(accountSid));
+                    map.put("extension_sid", DaoUtils.writeSid(extensionConfiguration.getSid()));
+
+                    if (extensionConfiguration.getConfigurationData() != null)
```

Comment:
@abdulazizali77 I believe that first thing we should do is to check that configuration data is not null and then proceed, if data is null we shouldn't attempt to do an insert in the DB. 
Also, the check should be done at ExtensionsConfigurationEndpoint for add and update operations